### PR TITLE
Quran router

### PR DESCRIPTION
**Pull request by nyzd**

### CODE REVIEW

File: src/models.rs [hunk 1]
```diff
@@ -54,3 +55,11 @@ pub struct NewToken<'a> {
     pub user_id: &'a i32,
     pub token_hash: &'a String,
 }
+
+#[derive(Queryable, PartialEq, Debug, Serialize)]
```
Comment (nyzd) [inline src/models.rs:59]: PartialEq

File: migrations/2022-10-27-100046_create_quran_text/up.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+-- Your SQL goes here
+
+CREATE TABLE quran_text (
+    id serial NOT NULL,
+    surah int4 NOT NULL DEFAULT 0,
```
Comment (nyzd) [inline migrations/2022-10-27-100046_create_quran_text/up.sql]: _id

File: migrations/2022-10-27-100046_create_quran_text/up.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+   	verse int4 NOT NULL DEFAULT 0,
```
Comment (nyzd) [inline migrations/2022-10-27-100046_create_quran_text/up.sql]: _number

File: src/routers/quran/quran.rs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+/// `/quran?from=1&to=10`
+#[get("/quran")]
+pub async fn quran(
+    query: web::Query<QuranQuery>,
+    pool: web::Data<DbPool>,
+) -> Result<HttpResponse, Error> {
+    use crate::schema::quran_text::dsl::*;
+
+    validate(&query.0)?;
+
```
Comment (nyzd) [inline src/routers/quran/quran.rs:27]: what if from is higher than to?

**Review by nyzd — COMMENTED** [review]
nema

File: src/routers/quran/quran.rs (hunk 4 above)
Comment (nyzd) [reply src/routers/quran/quran.rs:27]: TODO

**Review by al-abd — APPROVED** [review]

**Pull request merged**